### DISCUSSION TITLE
Errors while updating RPiR in R studio

### DISCUSSION THREAD

**Discussion by sreedevi-gummuluri** [IBAHCM/RPiR#420] [Archive] [answered]
@IBAHCM/rpir-instructors 

I receive the following error when I run "devtools::install_github("IBAHCM/RPiR")"

~~~~~~~~~~~
Using github PAT from envvar GITHUB_PAT
Error: Failed to install 'RPiR' from GitHub:
  HTTP error 401.
  Bad credentials

  Rate limit remaining: 58/60
  Rate limit reset at: 2021-11-24 20:23:48 UTC
~~~~~~~~~~~

"install_github("IBAHCM/RPiR")" returns the following error:
"devtools" and "usethis" are already installed.

~~~~~~~~~~~
Error in install_github("IBAHCM/RPiR") : 
  could not find function "install_github"
~~~~~~~~~~~

My credentials have been working for new project creation since using the gitcreds_set() command. However, these new errors are showing up now. I've restarted R but that didn't help.

I'd appreciate any guidance.

thank you,
Sreedevi

**Comment by richardreeve** [comment]
Hi @sreedevi-gummuluri - it's difficult to know what the first one is about without seeing more of what you did, but the second is just that although you have installed `devtools`, you haven't loaded it using `library(devtools)` - as a result `install_github()` isn't accessible yet. Either call `library(devtools)` first or call `devtools::install_github("IBAHCM/RPiR")` directly.

**Reply by richardreeve** [reply]
To look at what is going wrong with the first issue, I might try running `usethis::git_sitrep()` to see if anything obvious shows up - if it doesn't you could just paste the output into this discussion so we can have a look.

**Comment by sreedevi-gummuluri** [comment, marked as answer]
@IBAHCM/rpir-instructors  
devtools were already loaded with no errors. Here's the output from git_sitrep(). We had already updated the PAT information. So, I'm not sure how to address this.

~~~~~~~~~~
> usethis::git_sitrep()
Git config (global)
• Name: 'sreedevi-gummuluri'
• Email: '2500105G@student.gla.ac.uk'
• Global (user-level) gitignore file: <unset>
• Vaccinated: FALSE
ℹ See `?git_vaccinate` to learn more
ℹ Defaulting to 'https' Git protocol
• Default Git protocol: 'https'
• Default initial branch name: <unset>
GitHub
• Default GitHub host: 'https://github.com'
ℹ '/Users/sreedevigummuluri/.Renviron' defines environment variable(s):
  - GITHUB_PAT 
  This can prevent your PAT from being retrieved from the Git credential store.
ℹ If you are troubleshooting PAT problems, the root cause may be an old, invalid PAT defined in '/Users/sreedevigummuluri/.Renviron'.
• Call `edit_r_environ()` to edit that file.
ℹ For most use cases, it is better to NOT define the PAT in `.Renviron`.
  Instead, call `gitcreds::gitcreds_set()` to put the PAT into the Git credential store.
• Personal access token for 'https://github.com': '<discovered>'
x Can't get user information for this token.
  The token may no longer be valid or perhaps it lacks the 'user' scope.
Error in gh_process_response(raw): 
GitHub API error (401): 
Message: Bad credentials
Read more at https://docs.github.com/rest

Git repo for current project
ℹ No active usethis project
~~~~~~~~~~

**Reply by richardreeve** [reply]
So, two things:

- First, you obviously hadn't run `library(devtools)` since you last restarted R, otherwise you wouldn't have got that error, so please try that again - restart R and then load the library and then run the command all in one R session.
- Second, the message above suggests that you have a problem because you have edited the `.Renviron` file *and* used `gitcreds::gitcreds_set()`. This is just because you tried the first thing and then the second when Sonia identified that as a potential solution. To fix that, try to undo the first one, by calling `usethis::edit_r_environ()` and deleting the line that mentions the `GITHUB_PAT`. Save the file and then restart RStudio. Hopefully it will then work, but if not then please show the output from `usethis::git_sitrep()` again.

**Reply by sreedevi-gummuluri** [reply]
Thank you @richardreeve, clearing the GITHUB_PAT line from .Renviron fixed the issue.
